### PR TITLE
Add support of `phpcs.dist.xml` as well as `.phpcs.dist.xml`

### PR DESCRIPTION
**Pull request by olivervogel**
## Description

Provide a better editor experience. As files should have a corresponding extension, as this is often used to recognize the file type and derive corresponding features such as highlighting.

This has already been done by many libraries with a similar setup.

- [PHPUnit](https://github.com/sebastianbergmann/phpunit/issues/4650)
- [PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5609/files#diff-29c6e254ef0f520f55e849f06703cb1ec51d5b3eb8ad5342b04d530383025989R12)
- [phpstant](https://github.com/phpstan/phpstan/issues/6561)

## Suggested changelog entry

Added support of `phpcs.dist.xml` & `.phpcs.dist.xml` configuration files.

## Related issues/external references

None.

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [x] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [ ] Documentation improvement

## PR checklist

- [X] I have checked there is no other PR open for the same change.
- [X] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [X] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [ ] I have added tests to cover my changes.
- [X] I have verified that the code complies with the projects coding standards.
- [ ] [Required for new sniffs] I have added XML documentation for the sniff.

### CODE REVIEW

**Comment by stronk7** [comment]
Ref.: https://github.com/squizlabs/PHP_CodeSniffer/issues/3744

**Comment by jrfnl** [comment]
Thank you @olivervogel for your interest in contributing to PHPCS.

I'm not strongly against this change, but I'm also not convinced of the need for it for the following reasons:
1. There are already four configuration file names supported, including two using the `.xml` extension, so you can already use `phpcs.xml` for your distribution file and `.phpcs.xml` for local overload files and have editor support/syntax highlighting.
2. The `[.]phpcs.dist.xml` file name(s) are already supported via `--standard=phpcs.dist.xml`, so why would these need to be auto-discovered ?
  If it is about contributors to the project knowing how to run PHPCS, that is something to document in, for instance, a composer script.
3. In any decent editor, you can configure what extensions should be treated as what file type for the purpose of syntax highlighting and in most editors you can change this on the fly for each file with one click anyway, so again: why would this need support on the tooling level ?
4. I have seen little interest in this feature so far (five likes in a year is not convincing).

In other words, the current argumentation is weak and what other tools do is of little concern and is not strengthening your case. Argue the case for _this_ tool.

On a practical note: please do not open pull requests for something which needs discussion first. Please respect the CONTRIBUTING guide, in particular [the section about new features](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md#requestingsubmitting-new-features).

As for the PR:
* _If_ this would be accepted, it will only be accepted for the 4.0 release, not before.
* There are no tests covering this code/change.
* I think a discussion would be needed about the file precedence order as I don't think this is the right order (and no argument is made why it should be this order)

**Comment by jrfnl** [comment]
Closing for lack of response.
